File: kotlin-spring-data-jpa/src/test/kotlin/com/kotlin_spring_data_jpa/kotlin_spring_data_jpa/repositories/BeerRepositoryTest.java
package com.kotlin_spring_data_jpa.kotlin_spring_data_jpa.repositories;

import com.kotlin_spring_data_jpa.kotlin_spring_data_jpa.entities.Beer;
import com.kotlin_spring_data_jpa.kotlin_spring_data_jpa.models.BeerStyle;
import jakarta.validation.ConstraintViolationException;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;

import java.math.BigDecimal;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertThrows;

@DataJpaTest
public class BeerRepositoryTest {
    @Autowired
    BeerRepository beerRepository;

    @Test
    void testSaveBer() {
        Beer beer = new Beer();
        beer.setName("Beer Name");
        beer.setStyle(BeerStyle.ALE);
        beer.setUpc("Test UPC");
        beer.setPrice(new BigDecimal(123));

        Beer savedBeer = beerRepository.save(beer);
        beerRepository.flush();

        assertThat(savedBeer).isNotNull();
        assertThat(savedBeer.getId()).isNotNull();
        assertThat(savedBeer.getName()).isEqualTo(beer.getName());
    }

    @Test
    void testSaveBerWhereNameIsTooLong() {
        assertThrows(ConstraintViolationException.class, () -> {
            Beer beer = new Beer();
            beer.setName("Test Beer Test Beer Test Beer Test Beer Test Beer Test Beer");
            beer.setStyle(BeerStyle.ALE);
            beer.setUpc("Test UPC");
            beer.setPrice(new BigDecimal(123));

            Beer savedBeer = beerRepository.save(beer);
            beerRepository.flush();

            assertThat(savedBeer).isNotNull();
            assertThat(savedBeer.getId()).isNotNull();
            assertThat(savedBeer.getName()).isEqualTo(beer.getName());
        });
    }
}

File: kotlin-spring-r2dbc/src/test/kotlin/com/kotlin_spring_r2dbc/kotlin_spring_r2dbc/repositories/BeerRepositoryTest.kt
package com.kotlin_spring_r2dbc.kotlin_spring_r2dbc.repositories

import com.fasterxml.jackson.databind.ObjectMapper
import com.kotlin_spring_r2dbc.kotlin_spring_r2dbc.config.DatabaseConfig
import com.kotlin_spring_r2dbc.kotlin_spring_r2dbc.domain.Beer
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.data.r2dbc.DataR2dbcTest
import org.springframework.context.annotation.Import
import java.math.BigDecimal

@DataR2dbcTest
@Import(DatabaseConfig::class)
class BeerRepositoryTest {
    @Autowired
    lateinit var beerRepository: BeerRepository

     companion object {
         fun getTestBeer(): Beer =
             Beer(
                 name = "Space Dust",
                 style = "IPA",
                 price = BigDecimal.TEN,
                 quantity = 12,
                 upc = "123213"
             )
     }

    @Test
    fun testCreateJson() {
        val objectMapper = ObjectMapper()

        println(
            objectMapper.writeValueAsString(
                getTestBeer()
            )
        )
    }

    @Test
    fun saveNewBeer() {
        beerRepository.save(getTestBeer())
            .subscribe { beer -> println(beer.toString()) }
    }
}